packages/foam-vscode/src/features/folder-rename-handler.ts

```diff
@@ -0,0 +1,1112 @@
+                  connection.link,
+                  {
+                    target: newLinkContent,
+                  }
+                );
+                renameEdits.replace(
+                  actualSourceUri, // Edit is applied to the (potentially new) source file URI
+                  toVsCodeRange(edit.range),
+                  edit.newText
+                );
+                linksUpdatedForThisFile++;
+                Logger.debug(
+                  `  → Wikilink in "${vscode.workspace.asRelativePath(
+                    actualSourceUri
+                  )}" updated to: "${edit.newText}"`
+                );
+              }
+              break;
+            }
+            case 'link': {
+              // This is for standard markdown links [text](path)
+              const targetFoamUri = fromVsCodeUri(newFileUri); // URI of the moved file (link target)
+              const sourceDirectoryFoamUri =
+                fromVsCodeUri(actualSourceUri).getDirectory();
+
+              // Calculate the relative path string for the markdown link.
+              const newLinkPath = targetFoamUri.relativeTo(
+                sourceDirectoryFoamUri
+              ).path;
+
+              const edit = MarkdownLink.createUpdateLinkEdit(connection.link, {
+                target: newLinkPath, // This is the new href path
+              });
+              renameEdits.replace(
+                actualSourceUri, // Edit is applied to the (potentially new) source file URI
+                toVsCodeRange(edit.range),
+                edit.newText
+              );
+              linksUpdatedForThisFile++;
+              Logger.debug(
+                `  → Markdown link in "${vscode.workspace.asRelativePath(
+                  actualSourceUri
+                )}" updated to: "${edit.newText}"`
+              );
+              break;
+            }
+          }
+        }
+
+        // Now handle outgoing links from this moved file that may need relative path adjustment
+        const outgoingConnections = this.foam.graph.getLinks(
+          fromVsCodeUri(oldFileUri)
+        );
+
+        if (outgoingConnections.length > 0) {
+          Logger.debug(
+            `Found ${
+              outgoingConnections.length
+            } outgoing links from ${vscode.workspace.asRelativePath(
+              oldFileUri
+            )}`
+          );
+        }
+        for (const outgoingConnection of outgoingConnections) {
+          const { target } = MarkdownLink.analyzeLink(outgoingConnection.link);
+
+          Logger.debug(
+            `Checking outgoing link: "${outgoingConnection.link.rawText}" (type: ${outgoingConnection.link.type}, target: "${target}")`
+          );
+
+          // Process relative links that start with ../ for both regular links and wikilinks
+          // But skip links that point to files within the same renamed folder (those are handled by backlink processing)
+          if (!isAbsolute(target) && target.startsWith('../')) {
```

Comment:
Good point, I hadn't thought about this case - we should probably add a test case for this if you haven't already

(feels like a candidate for its own function)